### DISCUSSION TITLE
Cannot get Quali Lap Data?

### DISCUSSION THREAD

**Discussion by XBTJames** [theOehrly/Fast-F1#270] [Q&A] [answered]
Hi,

Not sure if there is something wrong I am doing here, but it seems I have no issue getting race lap data, but I cannot get any Quali lap data? I have included snippets of code below and their results:

Loading quali data gets me this:
Code:
quali = fastf1.get_session(2022, 'Singapore Grand Prix','Q')
quali.load()
print(quali.laps)

I get:
core           INFO 	Loading data for Singapore Grand Prix - Qualifying [v2.3.0]
api            INFO 	No cached data found for driver_info. Loading data...
api            INFO 	Fetching driver list...
api            INFO 	Data has been written to cache!
api            INFO 	No cached data found for timing_data. Loading data...
api            INFO 	Fetching timing data...
api            INFO 	Parsing timing data...
api            INFO 	Data has been written to cache!
api            INFO 	No cached data found for timing_app_data. Loading data...
api            INFO 	Fetching timing app data...
api            INFO 	Data has been written to cache!
core           INFO 	Processing timing data...
api            INFO 	No cached data found for session_status_data. Loading data...
api            INFO 	Fetching session status data...
api            INFO 	Data has been written to cache!
api            INFO 	No cached data found for track_status_data. Loading data...
api            INFO 	Fetching track status data...
api            INFO 	Data has been written to cache!
core        WARNING 	No lap data for driver 16
core        WARNING 	No lap data for driver 11
core        WARNING 	No lap data for driver 44
core        WARNING 	No lap data for driver 55
core        WARNING 	No lap data for driver 14
core        WARNING 	No lap data for driver 4
core        WARNING 	No lap data for driver 10
core        WARNING 	No lap data for driver 1
core        WARNING 	No lap data for driver 20
core        WARNING 	No lap data for driver 22
core        WARNING 	No lap data for driver 63
core        WARNING 	No lap data for driver 18
core        WARNING 	No lap data for driver 47
core        WARNING 	No lap data for driver 5
core        WARNING 	No lap data for driver 24
core        WARNING 	No lap data for driver 77
core        WARNING 	No lap data for driver 3
core        WARNING 	No lap data for driver 31
core        WARNING 	No lap data for driver 23
core        WARNING 	No lap data for driver 6
core        WARNING 	Failed to load lap data!
api            INFO 	No cached data found for car_data. Loading data...
api            INFO 	Fetching car data...
api            INFO 	Parsing car data...
api            INFO 	Data has been written to cache!
api            INFO 	No cached data found for position_data. Loading data...
api            INFO 	Fetching position data...
api            INFO 	Parsing position data...
api            INFO 	Data has been written to cache!
core        WARNING 	Failed to load telemetry data!
api            INFO 	No cached data found for weather_data. Loading data...
api            INFO 	Fetching weather data...
api            INFO 	Data has been written to cache!
api            INFO 	No cached data found for race_control_messages. Loading data...
api            INFO 	Fetching race control messages...
api            INFO 	Data has been written to cache!
core           INFO 	Finished loading data for 20 drivers: ['16', '11', '44', '55', '14', '4', '10', '1', '20', '22', '63', '18', '47', '5', '24', '77', '3', '31', '23', '6']

---------------------------------------------------------------------------
DataNotLoadedError                        Traceback (most recent call last)
/var/folders/8_/ypsv4hw13_36sgxpznxy4zj40000gn/T/ipykernel_773/128215762.py in <module>
      1 quali = fastf1.get_session(2022, 'Singapore Grand Prix','Q')
      2 quali.load(laps=True)
----> 3 print(quali.laps)

~/opt/anaconda3/lib/python3.9/site-packages/fastf1/core.py in laps(self)
   1050         Data is available after calling `Session.load` with ``laps=True``
   1051         """
-> 1052         return self._get_property_warn_not_loaded('_laps')
   1053 
   1054     @property

~/opt/anaconda3/lib/python3.9/site-packages/fastf1/core.py in _get_property_warn_not_loaded(self, name)
   1015         d = getattr(self, name, None)
   1016         if d is None:
-> 1017             raise DataNotLoadedError("The data you are trying to access has not "
   1018                                      "been loaded yet. See `Session.load`")
   1019         return d

DataNotLoadedError: The data you are trying to access has not been loaded yet. See `Session.load`

Is there a datasource I need to set up to get lap data for quali and practice sessions? It seems that race sessions load fine and I was able to replicate the "LEC vs HAM" chart on the github readme page.

edit: this seems to only affect some GPs. I haven't tested all of them but Singapore 2022, Saudi Arabia 2022, Australia 2022 do not work while Bahrain, Canada, Miami and Spain 2022 work as expected...

**Comment by theOehrly** [comment, marked as answer]
It's a bit surprising that this works for some Qualis but doesn't work for others, especially ones that are far in the past and have previously worked. Also, I cannot reproduce the problem myself. 
My first guess is that this might be a server-side error that has been fixed by now. Can you please just try this again and see if everything suddenly starts working? If it still doesn't work, please try to run the script without caching enabled to see if that maybe makes a difference.

**Reply by XBTJames** [reply]
It seems to be some sort of a caching issue, when I turn off my caching it loads totally fine. Today at around Noon UTC:
w/o caching:
api         WARNING 	

NO CACHE! Api caching has not been enabled! 
	It is highly recommended to enable this feature for much faster data loading!
	Use `fastf1.Cache.enable_cache('path/to/cache/')`

core           INFO 	Loading data for Singapore Grand Prix - Qualifying [v2.3.0]
api            INFO 	Fetching driver list...
api            INFO 	Fetching timing data...
api            INFO 	Parsing timing data...
api            INFO 	Fetching timing app data...
core           INFO 	Processing timing data...
api            INFO 	Fetching session status data...
api            INFO 	Fetching track status data...
api            INFO 	Fetching car data...
api            INFO 	Parsing car data...
api            INFO 	Fetching position data...
api            INFO 	Parsing position data...
api            INFO 	Fetching weather data...
api            INFO 	Fetching race control messages...
core           INFO 	Finished loading data for 20 drivers: ['16', '11', '44', '55', '14', '4', '10', '1', '20', '22', '63', '18', '47', '5', '24', '77', '3', '31', '23', '6']

if I restart my kernel, delete any existing Singapore GP data from the cache, and enable caching, I get the same error as above. Should be fine with no caching, but very strange.

**Reply by theOehrly** [reply]
Ok, interesting but very strange. At least the problem is reproducible for you. 
Can you describe your environment in more detail please, so I can try to figure out what's going on? You say kernel, so I assume Jupyter?
Which version, python version, os and so on do you use?

**Reply by XBTJames** [reply]
Yes-  I am running in Jupyter Notebook, Python 3.9.7

**Reply by XBTJames** [reply]
Hmm I tried clearing all my cache again and enabling caching and I can no longer get it to replicate the errors.

**Reply by theOehrly** [reply]
How did you clear the cache? By hand or using the builtin function for that?
